gui/layers/_markers_layer.py

```diff
@@ -129,31 +156,31 @@ def size(self) -> Union[int, float, np.ndarray, list]:
         """float, ndarray: size of the marker symbol in px
         """
 
-        return self._size
+        return self._size_original
 
     @size.setter
     def size(self, size: Union[int, float, np.ndarray, list]) -> None:
 
-        if np.isscalar(size):
-            self._size = size
-
-            self.refresh()
-
-        elif isinstance(size, Iterable):
-            assert len(size) == len(self._coords), \
-             'If size is a list/array, must be the same length as '\
+        size_array = np.array(size)
+        if len(size_array.shape) == 2:
+            assert(size_array.shape == self._coords.shape), \
+             'If size is a 2D array, must be the same shape as '\
              'coords'
-
-            if isinstance(size, list):
-                self._size = np.asarray(size)
-
+            self._size = size_array
+        elif len(size_array.shape) == 1:
+            assert (len(size_array) == len(self._coords) or  len(size_array) == self.ndim), \
```

Comment:
Oh man. This is bad. What if I have 2 markers in a 2D canvas and I want to set the size of the two markers?

imho we shouldn't do this much magic here and we might have to live with giving a column vector for the sizes. Or some other unambiguous solution...

Reply:
ah yes i was worried about this ambiguous case ...

I think that just trying to do the broadcast, then catching an error if there is one and trying the ambiguous thing (making the column vector and broadcasting) and then catching that error and returning a value error seems like the way to go.

I'll implement now and we can have a look if we like it. If not I'm ok going down the road of only broadcasting and forcing people to pass a column vector if they want that special behavior.